### PR TITLE
feat(api-idorslug): Created `api.id-or-slug-enabled` Option

### PR DESCRIPTION
**Pull request by iamrajjoshi**
Created  `api.id-or-slug-enabled` option for supporting id and slug for our path parameters

For: https://github.com/getsentry/team-core-product-foundations/issues/194

### CODE REVIEW

File: src/sentry/features/temporary.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ def register_temporary_features(manager: FeatureManager):
     manager.add("organizations:anr-analyze-frames", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
     manager.add("organizations:anr-improvements", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
     manager.add("organizations:api-auth-provider", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
+    manager.add("organizations:api-id-or-slug-support", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
```
Comment (cathteng) [inline src/sentry/features/temporary.py:49]: should we roll this out on flagr or some other method?

Comment (iamrajjoshi) [reply src/sentry/features/temporary.py]: We will be using an option now!

File: src/sentry/options/defaults.py [hunk 2]
```diff
@@ -264,6 +264,15 @@
     flags=FLAG_ALLOW_EMPTY | FLAG_PRIORITIZE_DISK | FLAG_REQUIRED,
 )
 
+# API
+# Killswitch for apis to work with id or slug as path parameters
+register(
+    "api.id-or-slug-enabled",
+    default=False,
+    type=Bool,
+    flags=FLAG_MODIFIABLE_BOOL | FLAG_AUTOMATOR_MODIFIABLE,
```
Comment (iamrajjoshi) [inline src/sentry/options/defaults.py:273]: We want this flag to be modifiable modified live.

Comment (iamrajjoshi) [reply src/sentry/options/defaults.py:273]: Just had an offline conversation with Filippo, we do not need the `FLAG_PRIORITIZE_DISK` flag, `FLAG_AUTOMATOR_MODIFIABLE` is enough!

**Review by schew2381 — APPROVED** [review]

**Pull request merged**